File: app/src/main/java/com/mss/imagesearcher/di/AppComponent.java
package com.mss.imagesearcher.di;


import androidx.annotation.NonNull;

import com.mss.imagesearcher.view.detail.DetailActivity;
import com.mss.imagesearcher.view.imagelist.ImageListAdapter;
import com.mss.imagesearcher.view.info.InfoActivity;
import com.mss.imagesearcher.view.main.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, DBModule.class, NetworkModule.class, FilesModule.class})
public interface AppComponent {

    void inject(@NonNull final MainActivity mainActivity);

    void inject(@NonNull final DetailActivity detailActivity);

    void inject(@NonNull final ImageListAdapter imageListAdapter);

    void inject(@NonNull final InfoActivity infoActivity);
}

File: app/src/main/java/com/mss/imagesearcher/di/AppComponent.kt
package com.mss.imagesearcher.di


import com.mss.imagesearcher.view.detail.DetailFragment
import com.mss.imagesearcher.view.history.HistoryFragment
import com.mss.imagesearcher.view.imagelist.ImageListAdapter
import com.mss.imagesearcher.view.imagelist.ImageListFragment
import com.mss.imagesearcher.view.info.InfoFragment
import com.mss.imagesearcher.view.main.MainActivity
import com.mss.imagesearcher.view.settings.SettingsFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, NetworkModule::class, FilesModule::class, DBModule::class])
interface AppComponent {

    fun inject(mainActivity: MainActivity)

    fun inject(detailFragment: DetailFragment)

    fun inject(imageListAdapter: ImageListAdapter)

    fun inject(infoFragment: InfoFragment)

    fun inject(imageListFragment: ImageListFragment)

    fun inject(historyFragment: HistoryFragment)

    fun inject(settingsFragment: SettingsFragment)
}
